src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyCacheableRequest.java

```diff
@@ -472,6 +480,30 @@ private void incAttempts()
         attempts++;
     }
 
+    private Consumer<ListFW.Builder<HttpHeaderFW.Builder, HttpHeaderFW>> mutateRequestHeaders(
+        ListFW<HttpHeaderFW> requestHeaders)
+    {
+        return (ListFW.Builder<HttpHeaderFW.Builder, HttpHeaderFW> builder) ->
+        {
+           requestHeaders.forEach(h ->
+           {
+               final String name = h.name().asString();
+               final String value = h.value().asString();
+               if(!CONTENT_LENGTH.equalsIgnoreCase(name) &&
+                  !AUTHORIZATION.equals(name))
```

Comment:
Headers should be guaranteed lowercase by `http` and `http2` nukleus, so you should be able to compare the `StringFW` without calling `asString()`, avoiding unnecessary UTF-8 decoding and garbage.

Reply:
Done.